nifi-nar-bundles/nifi-snmp-bundle/nifi-snmp-processors/src/main/java/org/apache/nifi/snmp/processors/properties/V1TrapProperties.java

```diff
@@ -0,0 +1,118 @@
+            "A linkDown trap signifies that the sending protocol entity recognizes a failure in one of " +
+                    "the communication links represented in the agent's configuration.");
+
+    private static final AllowableValue GENERIC_TRAP_TYPE_LINK_UP = new AllowableValue("3", "Link Up",
+            "A linkUp trap signifies that the sending protocol entity recognizes that one of the communication " +
+                    "links represented in the agent's configuration has come up.");
+
+    private static final AllowableValue GENERIC_TRAP_TYPE_AUTH_FAILURE = new AllowableValue("4", "Authentication Failure",
+            "An authenticationFailure trap signifies that the sending protocol entity is the addressee of a " +
+                    "protocol message that is not properly authenticated.  While implementations of the SNMP must be " +
+                    "capable of generating this trap, they must also be capable of suppressing the emission of such traps " +
+                    "via an implementation- specific mechanism.");
+
+    private static final AllowableValue GENERIC_TRAP_TYPE_EGP_NEIGHBOR_LOSS = new AllowableValue("5", "EGP Neighbor Loss",
+            "An egpNeighborLoss trap signifies that an EGP neighbor for whom the sending protocol entity was " +
+                    "an EGP peer has been marked down and the peer relationship no longer obtains.");
+
+    private static final AllowableValue GENERIC_TRAP_TYPE_ENTERPRISE_SPECIFIC = new AllowableValue("6", "Enterprise Specific",
+            "An enterpriseSpecific trap signifies that a particular enterprise-specific trap has occurred which " +
+                    "can be defined in the Specific Trap Type field.");
+
+    public static final PropertyDescriptor ENTERPRISE_OID = new PropertyDescriptor.Builder()
+            .name("snmp-trap-enterprise-oid")
+            .displayName("Enterprise OID")
+            .description("Enterprise is the vendor identification (OID) for the network management sub-system that generated the trap.")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .dependsOn(SNMP_VERSION, SNMP_V1)
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+            .build();
+
+    public static final PropertyDescriptor AGENT_ADDRESS = new PropertyDescriptor.Builder()
+            .name("snmp-trap-agent-address")
+            .displayName("SNMP Trap Agent Address")
+            .description("The address where the SNMP Manager sends the trap.")
+            .required(true)
+            .defaultValue("0")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .dependsOn(SNMP_VERSION, SNMP_V1)
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+            .build();
+
+    public static final PropertyDescriptor PREDEFINED_GENERIC_TRAP_TYPE = new PropertyDescriptor.Builder()
```

Comment:
Having both a drop-down list and a text-box for the same purpose is more confusing than convenient in my opinion.